pkg/gui/services/custom_commands/session_state_loader.go

```diff
@@ -181,19 +182,30 @@ type SessionState struct {
 }
 
 func (self *SessionStateLoader) call() *SessionState {
+	var commit = commitShimFromModelCommit(self.c.Contexts().LocalCommits.GetSelected())
+
+	if commit == nil {
+		commit = commitShimFromModelCommit(self.c.Contexts().ReflogCommits.GetSelected())
+
+		if commit == nil {
+			commit = commitShimFromModelCommit(self.c.Contexts().SubCommits.GetSelected())
+		}
+	}
+
 	return &SessionState{
 		SelectedFile:           fileShimFromModelFile(self.c.Contexts().Files.GetSelectedFile()),
 		SelectedPath:           self.c.Contexts().Files.GetSelectedPath(),
-		SelectedLocalCommit:    commitShimFromModelCommit(self.c.Contexts().LocalCommits.GetSelected()),
-		SelectedReflogCommit:   commitShimFromModelCommit(self.c.Contexts().ReflogCommits.GetSelected()),
+		SelectedLocalCommit:    commit,
+		SelectedReflogCommit:   commit,
+		SelectedCommit:         commit,
```

Comment:
This is backwards. We need to keep setting SelectedLocalCommit, SelectedReflogCommit, and SelectedSubCommit to what they were set to before, for backwards compatibility. Only SelectedCommit will point to one of them, depending on context; see above.